spring-kafka/src/main/java/org/springframework/kafka/support/Acknowledgment.java

```diff
@@ -57,8 +74,25 @@ default void nack(long sleepMillis) {
 	 * @param sleepMillis the time to sleep in milliseconds; the actual sleep time will be larger
 	 * of this value and the container's {@code maxPollInterval}, which defaults to 5 seconds.
 	 * @since 2.3
+	 * @deprecated in favor of {@link #nack(int, Duration)}
 	 */
+	@Deprecated(since = "2.8.7", forRemoval = true)
```

Comment:
For backports, `since` and `forRemoval` will be a problem. On the other hand, IDE generated a warning without them in 3.x. They could be dropped during backporting or removed also in master. WDYT?

Reply:
We should make a separate PR for 2.8.x and actually remove the method in 3.0.

Reply:
I created #2298 which should be JDK 8+ compatible and remove the methods for 3.0.